### PR TITLE
File-Upload: Molecule Component

### PR DESCRIPTION
**Pull request by salilkapoor**

### CODE REVIEW

File: lib/components/molecules/Form/FieldFileInput/FieldFileInput.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  name,
+  label,
+  onChange,
+  onBlur,
+  labelProps,
+  fieldProps,
+  ...others
+}: Props): Node => (
+  // TODO: Add support for label position
+  <div className={className}>
```
Comment (vinodloha) [inline lib/components/molecules/Form/FieldFileInput/FieldFileInput.js]: Does it support drag and drop based upload in a container? That was one of the req too

File: lib/components/molecules/Form/FieldFileInput/tests/FieldFileInput.test.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import { shallow, render } from 'enzyme';
+import { FieldFileInput } from '../FieldFileInput';
+
+describe('FieldInput Component', () => {
```
Comment (vinodloha) [inline lib/components/molecules/Form/FieldFileInput/tests/FieldFileInput.test.js:6]: These tests are good start but not enough, ask your self does it cover every line of code you have written in component?

File: lib/components/molecules/Form/Form.js [hunk 3]
```diff
@@ -10,6 +10,7 @@ import FieldSelect from './FieldSelect';
 import FieldTextarea from './FieldTextarea';
 import FieldRadioGroup from './FieldRadioGroup';
 import FieldCheckbox from './FieldCheckbox';
+import FieldFileInput from './FieldFileInput';
```
Comment (vinodloha) [inline lib/components/molecules/Form/Form.js:13]: Nice

File: lib/components/molecules/Form/FieldFileInput/FieldFileInput.js [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    <Field
+      id={id}
+      name={name}
+      type={type}
+      render={({ field, form }) => {
+        const { handleChange, handleBlur } = form;
+        return (
+          <Input
```
Comment (ankurkakroo2) [inline lib/components/molecules/Form/FieldFileInput/FieldFileInput.js]: Think of it this way.
Where is the CSS for input really residing ? 
FieldInput is just a wrapper and should have CSS only for label  and outer styling.

You have 2 options:
Write CSS in Input.style.js for [type=file]

or better: create a new component named FileInput, Use <Input type="file"> inside it and write css in FileInput.style.js

And then wrap everything in this component

Reach out to me if you need further clarification.

File: lib/components/molecules/Form/FieldFileInput/FieldFileInput.js [hunk 5]
```diff
@@ -0,0 +1,84 @@
+// @flow
```
Comment (ankurkakroo2) [inline lib/components/molecules/Form/FieldFileInput/FieldFileInput.js:1]: Can you attach screenshots in your PR ?

File: lib/components/molecules/Form/FieldFileInput/FieldFileInput.style.js [hunk 6]
```diff
@@ -0,0 +1,18 @@
+import { css } from 'styled-components';
+
+// TODO: Fix the duplicate styling in input.style.js
+const styles = css`
+  margin-bottom: 0.75rem;
```
Comment (ankurkakroo2) [inline lib/components/molecules/Form/FieldFileInput/FieldFileInput.style.js]: We should be using margins and paddings from variables

File: lib/components/molecules/Form/FieldFileInput/FieldFileInput.js (hunk 5 above)
Comment (salilkapoor) [reply lib/components/molecules/Form/FieldFileInput/FieldFileInput.js:1]: image

File: lib/components/molecules/Form/FieldFileInput/FieldFileInput.style.js (hunk 6 above)
Comment (salilkapoor) [reply lib/components/molecules/Form/FieldFileInput/FieldFileInput.style.js]: I changed to variabled for colors, spacing wherever applicable.

File: lib/components/molecules/Form/FieldFileInput/FieldFileInput.js (hunk 4 above)
Comment (salilkapoor) [reply lib/components/molecules/Form/FieldFileInput/FieldFileInput.js]: I followed second approach as suggested. Also, it was giving me problem in one more case i.e. value field as well, when i tried uploaded file and it was not able to set explicit value prop.

File: lib/components/molecules/Form/FieldFileInput/FieldFileInput.js (hunk 1 above)
Comment (salilkapoor) [reply lib/components/molecules/Form/FieldFileInput/FieldFileInput.js]: Implemented.

**Pull request closed without merging**